### ISSUE TITLE
Usability issue: clobbered runs, lost data

### ISSUE THREAD

**Issue by dandelionmane** [tensorflow/tensorboard#544]
It's very natural to write code that writes multiple runs to the same run directory, e.g. because you have code that explicitly specifies a log directory, and you run that code multiple times. Empirically, it looks like TensorBoard silently discards some of the data so as to show only one run (?)

It would be great to solve this problem for our users. There are basically two approaches:

1. Change TensorBoard so that it automatically detects when the run has restarted, and separates it into a new run (probably with a numerical sequential naming scheme). 
1.1 Ideally, add on some API that makes it natural to give these runs names or choose to preserve certain runs. (We've talked about adding explicit hyperparameter support to TensorBoard, I think that would help here.)

2. Add a new SummaryWriterManager type API, which when used in lieu of the summary writer automatically creates a clean directory structure for TensorBoard.

I think 1. is much preferable to 2. because there are so many SummaryWriters/FileWriters already present in many different libraries that use TensorFlow, it would be difficult to get an API usage shift at this time. 

@jart is this something you could include in your backend data ingestion refactoring?

**Comment by chihuahua** [comment]
So would 1 way this issue manifest be when a user re-runs a model with a different learning rate or layer size, and then data from the previous run is discarded by [TensorBoard's purging logic](https://github.com/tensorflow/tensorboard/blob/c1be95b8cac849759b20100401c94077f9bcaaf4/tensorboard/backend/event_processing/event_accumulator.py#L338)?

**Comment by dandelionmane** [comment]
Yup, exactly!

**Comment by bileschi** [comment]
This may be interesting as we revisit the summary writing api
